flux-kube/etc/01-flux-kube-daemon

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Submit flux kube daemon as a job 
+# so it is cleaned up when flux exits
+if test $(flux getattr rank) -eq 0; then
+    flux mini submit flux kube daemonize
```

Comment:
Would it be worth alloc-bypassing this as well?

Reply:
I tried that via:
```bash
if test $(flux getattr rank) -eq 0; then
    flux jobtap load alloc-bypass.so
    flux mini submit \
         --setattr=system.alloc-bypass.R="$(flux kvs get resource.R)" \
         flux kube daemonize
fi
```
which behaves exactly the same as when it isn't alloc-bypassed.

Reply:
This is a use case we've been calling a "service job" (https://github.com/flux-framework/flux-core/issues/3842) and isn't supported quite as easily as we'd like.

In this case by using alloc-bypass the difference is that a core isn't allocated from the instance, instead `flux kube daemonize` is run as a job that takes no resources.

You probably don't want to run `flux kube daemonize` across _all_ resources of the instance. You can perhaps just run it on the resources of rank 0, `flux kvs get resource.R | flux R decode --include=0`

We'll come back around once we have a better way to do this.

Reply:
I think I fixed this via @grondo 's suggestion.